### ISSUE TITLE
"quote" option to allow single quotes in template literals

### ISSUE THREAD

**Issue by cpojer** [eslint/eslint#5866]
**What version of ESLint are you using?**
2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

At Facebook we are using the "quote" option like this: `'quotes': [WARNING, 'single', 'avoid-escape']`

We sometimes write strings with backticks, like:

```js
`this doesn't use single quotes because it would require a back slash`
```

The quote rule will trigger a warning in this case and attempt to replace it to:

```js
'this doesn\'t use single quotes because it would require a back slash'
```

The 'avoid-escape' option leads me to believe that the lint rule shouldn't be triggered, so this seems like a bug to me.

**Comment by hzoo** [comment]
I think this is covered by https://github.com/eslint/eslint/issues/5234 and fixed in https://github.com/eslint/eslint/pull/5661 but not released yet.

**Comment by cpojer** [comment]
That's nice! Is there a new release coming soon? :)

**Comment by ilyavolodin** [comment]
@cpojer We try to do releases every 2 weeks, so another one coming soon.
